web/src/main/java/org/openmrs/web/Listener.java

```diff
@@ -258,29 +259,34 @@ public void contextInitialized(ServletContextEvent event) {
 			log.error(MarkerFactory.getMarker("FATAL"), "Failed to obtain JDBC connection", e);
 		}
 	}
-	
-	private void loadCsrfGuardProperties(ServletContext servletContext) throws FileNotFoundException, IOException {	
-		File file = new File(OpenmrsUtil.getApplicationDataDirectory(), "csrfguard.properties");
-		InputStream inputStream = null;
+
+	private void loadCsrfGuardProperties(ServletContext servletContext) throws IOException {
+		File csrfGuardFile = new File(OpenmrsUtil.getApplicationDataDirectory(), "csrfguard.properties");
+		InputStream csrfGuardInputStream = null;
+		Properties csrfGuardProperties = new Properties();
 		try {
-			inputStream = new FileInputStream(file);
-		}
-		catch (FileNotFoundException ex) {
+			csrfGuardInputStream = Files.newInputStream(csrfGuardFile.toPath());
+		} 
+		catch (IOException ioException) {
 			final String fileName = servletContext.getRealPath("/WEB-INF/csrfguard.properties");
-			inputStream = new FileInputStream(fileName);
-			OutputStream outputStream = new FileOutputStream(file);
-			IOUtils.copy(inputStream, outputStream);
-		    IOUtils.closeQuietly(outputStream);
-		    IOUtils.closeQuietly(inputStream);
-		    
-		    //Moved to EOF by the copy operation. So open it again.
-		    inputStream = new FileInputStream(file);
-		}
-		Properties properties = new Properties();
-		properties.load(inputStream);
-		IOUtils.closeQuietly(inputStream);
-		CsrfGuard.load(properties);
-		
+			try {
+				csrfGuardInputStream = Files.newInputStream(Paths.get(fileName));
+			} 
+			catch (IOException ioException1) {
+				ioException1.printStackTrace();
+				throw ioException1;
+			}
+		}
+		csrfGuardProperties.load(csrfGuardInputStream);
+		Properties runtimeProperties = OpenmrsUtil.getRuntimeProperties(WebConstants.WEBAPP_NAME);
+		runtimeProperties.stringPropertyNames().forEach(property -> {
+			if(property.startsWith("org.owasp.csrfguard")) {
+				csrfGuardProperties.setProperty(property, runtimeProperties.getProperty(property));
+			}
+		});		
+		IOUtils.closeQuietly(csrfGuardInputStream);
```

Comment:
This really needs to be done in a finally block. There are way to many ways an exception can be thrown in the above code and so the IO stream never closed.